test/e2e/storage/drivers/csi.go

```diff
@@ -602,6 +606,26 @@ func (g *gcePDCSIDriver) PrepareTest(f *framework.Framework) (*testsuites.PerTes
 	}, cleanupFunc
 }
 
+func WaitForPDCSIController(driverNamespace *v1.Namespace, cs clientset.Interface) error {
+	const ssName = "csi-gce-pd-controller"  // Taken from testing-manifests/storage-csi/gce-pd/controller_ss.yaml.
+	const csiPDDriverDeployTimeout = 5 * time.Minute
+
+	waitErr := wait.PollImmediate(10*time.Second, csiPDDriverDeployTimeout, func() (bool, error) {
+		_, err := cs.AppsV1().StatefulSets(driverNamespace.Name).Get(context.TODO(), ssName, metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		if err != nil {  // apierrors.IsNotFound(err)
+			return false, nil
+		}
+		return true, nil
```

Comment:
This only waits for the statefulset object to exist, which should be immediately true once the create operation returns.

Do we want to wait for it and/or all the pods to be ready?

Reply:
Ah, that's my k8s ignorance. Yes, I think we should wait for the pods.

I can dig to figure out what pods should exist from a SS but if you have a pointer handy so much the better.

Reply:
statefulset has `readyReplicas` that seem suitable: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#statefulset-v1-apps

Reply:
Thx for the pointer, used that field.

I was a little paranoid in the ready check, LMK if that's excessive.